### PR TITLE
STSMACOM-326: Custom fields: Multi-select dropdown

### PR DESCRIPTION
**Pull request by Vladyslav-Velytskyi**
## Description

- Add possibility to add Single select dropdown to custom fields

## Issue
[STSMACOM-326](https://issues.folio.org/browse/STSMACOM-326)

## TODOS
- Code will be implemented in [STSMACOM-330](https://issues.folio.org/browse/STSMACOM-330)
- Tests will be implemented in [STSMACOM-331](https://issues.folio.org/browse/STSMACOM-331)
- Logic of handling radio button and select options will be significantly simplified after the backend provides a unique id for each option

#### Edit Page
image

#### View Page
image

### CODE REVIEW

File: lib/CustomFields/components/FieldAccordion/edit-fields/shared-fields/OptionsField.js [hunk 1]
```diff
@@ -66,13 +70,55 @@ const OptionsField = ({
...
+        validate={validateLabel}
+        validateFields={[]}
+        marginBottom0
+      >
+        {(fieldProps) => (
+          <TextField
+            {...fieldProps}
+            onChange={e => {
+              if (isMultiSelect && defaultOptions.includes(fieldProps.input.value)) {
+                changeFieldValue(`${fieldNamePrefix}.selectField.defaults`, [...defaultOptions, e.target.value]);
```
Comment (MaksymDryha) [inline lib/CustomFields/components/FieldAccordion/edit-fields/shared-fields/OptionsField.js:100]: Shouldn't we change the option at `rowData.rowIndex` position specifically?

File: lib/CustomFields/pages/EditCustomFieldsSettings/EditCustomFieldsSettings.js [hunk 2]
```diff
@@ -192,6 +192,17 @@ const EditCustomFieldsSettings = ({
...
       }
 
+      if (customFieldData.type === fieldTypes.MULTISELECT) {
+        customFieldData.selectField.defaults = customFieldData.selectField.defaults
+          .reduce((acc, option) => {
+            if (!customFieldData.selectField.options.values.includes(option)) {
+              return acc;
+            }
+
+            return Array.isArray(option) ? [...acc, ...option] : [...acc, option];
```
[the remark this answers is not in the document]
Comment (Vladyslav-Velytskyi) [reply lib/CustomFields/pages/EditCustomFieldsSettings/EditCustomFieldsSettings.js:202]: Yes, when we click on checkbox at first, option becomes an array. I check this issue many times

File: lib/CustomFields/components/FieldAccordion/edit-fields/shared-fields/OptionsField.js [hunk 3]
```diff
@@ -93,40 +139,34 @@ const OptionsField = ({
...
-                          ? fields.value[rowData.rowIndex - 1]
-                          : NO_DEFAULT_OPTIONS_VALUE
-                        }
-                        marginBottom0
-                        type="radio"
-                        className={css.defaultControl}
-                        disabled={fields.value[rowData.rowIndex - 1] === ''}
-                      />
+                    label: rowData => getLabelField(rowData),
+                    defaults: rowData => (isMultiSelect ?
```
[the remark this answers is not in the document]
Comment (Vladyslav-Velytskyi) [reply lib/CustomFields/components/FieldAccordion/edit-fields/shared-fields/OptionsField.js:143]: Actually it needs, when arrow function returns a condition we need to cover it in round brackets
(eslint rule - no-confusing-arrow)

File: lib/CustomFields/pages/EditCustomFieldsSettings/EditCustomFieldsSettings.js [hunk 4]
```diff
@@ -192,6 +192,17 @@ const EditCustomFieldsSettings = ({
         }
       }
 
+      if (customFieldData.type === fieldTypes.MULTISELECT) {
+        customFieldData.selectField.defaults = customFieldData.selectField.defaults
+          .reduce((acc, option) => {
+            if (!customFieldData.selectField.options.values.includes(option)) {
```
[the remark this answers is not in the document]
Comment (Vladyslav-Velytskyi) [reply lib/CustomFields/pages/EditCustomFieldsSettings/EditCustomFieldsSettings.js:198]: Yes, when we change the name option we need to update it in default's array

File: lib/CustomFields/components/FieldAccordion/edit-fields/shared-fields/OptionsField.js (hunk 1 above)
Comment (Vladyslav-Velytskyi) [reply lib/CustomFields/components/FieldAccordion/edit-fields/shared-fields/OptionsField.js:100]: No, the default's array is not connected in any way with option fields and their indexes aren't equal

**Pull request merged**